src/commands/add/internal/input.ts

```diff
@@ -117,6 +123,44 @@ async function getAndValidateTargetName(
   return targetName
 }
 
+async function getAndValidateUpdateExisting(
+  targetScope: TargetScope,
+  targetName: string
+): Promise<{ targetJson: TargetJson; retry: boolean }> {
+  let config, targetJson: TargetJson
+  if (targetScope === TargetScope.Local) {
+    config = await getLocalConfig()
+  } else {
+    config = await getGlobalRcFile()
+  }
+
+  targetJson = config?.targets?.find((t: Target) => t.name === targetName)
+
+  if (targetJson) {
+    process.logger?.info(`Found target with same name: '${targetName}'`)
+    const errorMessage = 'Target update choice must be either 1 or 2.'
+    const choice = await getChoice(
+      'Please pick an option for the current target: ',
+      'Please choose either option 1 or 2.',
+      [
+        { title: '1. Update this target', value: 1 },
+        { title: '2. Go back and create a new target', value: 2 }
+      ]
+    ).catch(() => {
+      throw new Error(errorMessage)
+    })
+
+    if (choice === null || choice === undefined || Number.isNaN(choice)) {
```

Comment:
maybe we can just check `if (!choice)...`

Reply:
fixed